ui/src/state/diary/diary.ts

```diff
@@ -74,314 +65,319 @@ function diaryNoteDiff(flag: DiaryFlag, time: string, delta: NoteDelta) {
+      await new Promise<void>((resolve, reject) => {
+        api.poke({
           app: 'diary',
-          path: `/diary/${flag}/perm`,
-        });
-        get().batchSet((draft) => {
-          const diary = { perms, view: 'list' as DiaryDisplayMode };
-          draft.shelf[flag] = diary;
-          draft.diarySubs.push(flag);
+          mark: 'diary-create',
+          json: req,
+          onError: () => reject(),
+          onSuccess: async () => {
+            await useSubscriptionState.getState().track('diary/ui', (event) => {
+              const { update, flag } = event;
+              if (
+                'create' in update.diff &&
+                flag === `${req.group.split('/')[0]}/${req.name}`
+              ) {
+                return true;
+              }
+              return false;
+            });
+            resolve();
+          },
         });
-
-        await makeNotesStore(
-          flag,
-          get,
-          `/diary/${flag}/notes`,
-          `/diary/${flag}/ui`
-        ).initialize();
-      },
-    }),
-    {
-      name: createStorageKey('diary'),
-      version: storageVersion,
-      migrate: clearStorageMigration,
-      partialize: ({ shelf }) => ({
-        shelf,
-      }),
-    }
-  )
+      });
+    },
+    addSects: async (flag, sects) => {
+      await api.poke(diaryAction(flag, { 'add-sects': sects }));
+      const perms = await api.scry<DiaryPerm>({
+        app: 'diary',
+        path: `/diary/${flag}/perm`,
+      });
+      get().batchSet((draft) => {
+        draft.shelf[flag].perms = perms;
+      });
+    },
+    delSects: async (flag, sects) => {
+      await api.poke(diaryAction(flag, { 'del-sects': sects }));
+      const perms = await api.scry<DiaryPerm>({
+        app: 'diary',
+        path: `/diary/${flag}/perm`,
+      });
+      get().batchSet((draft) => {
+        draft.shelf[flag].perms = perms;
+      });
+    },
+    getOlderNotes: async (flag: string, count: number) => {
+      await makeNotesStore(
+        flag,
+        get,
+        `/diary/${flag}/notes`,
+        `/diary/${flag}/ui`
+      ).getOlder(count.toString());
+    },
+    getNewerNotes: async (flag: string, count: number) => {
+      await makeNotesStore(
+        flag,
+        get,
+        `/diary/${flag}/notes`,
+        `/diary/${flag}/ui`
+      ).getNewer(count.toString());
+    },
+    initialize: async (flag) => {
+      if (get().diarySubs.includes(flag)) {
+        return;
+      }
+
+      const perms = await api.scry<DiaryPerm>({
+        app: 'diary',
+        path: `/diary/${flag}/perm`,
+      });
+      get().batchSet((draft) => {
+        const diary = { perms, view: 'list' as DiaryDisplayMode };
+        draft.shelf[flag] = diary;
+        draft.diarySubs.push(flag);
+      });
+
+      await makeNotesStore(
+        flag,
+        get,
+        `/diary/${flag}/notes`,
+        `/diary/${flag}/ui`
+      ).initialize();
+    },
+  }),
+  ['briefs', 'shelf', 'notes'],
```

Comment:
A little concerned that notes is going to balloon our local storage usage even more. One thing that would be nice is if we could use the partialize function from zustand here to only pull outlines instead of full notes.

Reply:
You're right, will make a separate issue for this.